code/datums/gamemode/factions/bloodcult/bloodcult.dm

```diff
@@ -78,12 +105,26 @@
 
 /datum/faction/bloodcult/AdminPanelEntry(var/datum/admins/A)
 	var/list/dat = ..()
-	// TODO UPHEAVAL PART 2, admin debug buttons
+	
+	dat += "<br>"
+	dat += "<a href='?src=\ref[src];unlockRitual=1'>\[Unlock Ritual\]</A><br>"
+	dat += "<br>"
+
 	return dat
 
 /datum/faction/bloodcult/Topic(href, href_list)
 	..()
-	// TODO UPHEAVAL PART 2, admin debug buttons
+
+	if(!usr.check_rights(R_ADMIN))
+		message_admins("[usr] tried to access bloodcult faction Topic() without permissions.")
+		return
+
+	if(href_list["unlockRitual"])
+		var/datum/bloodcult_ritual/R = input(usr,"Select a ritual to unlock.", "Unlock", null) as null|anything in locked_rituals
+		if(R)
+			R.Unlock(TRUE)
+			locked_rituals -= R
```

Comment:
This should be done inside `Unlock()`, no?

Reply:
It wasn't done inside Unlock(). Now it is.